File: frontend/src/app/services/project.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Project, ProjectRequest } from '../models/models';
import { environment } from '../../environments/environment';

@Injectable({ providedIn: 'root' })
export class ProjectService {
  private url = `${environment.apiBaseUrl}/projects`;

  constructor(private http: HttpClient) {}

  getAll(): Observable<Project[]> {
    return this.http.get<Project[]>(this.url);
  }

  getById(id: string): Observable<Project> {
    return this.http.get<Project>(`${this.url}/${id}`);
  }

  create(request: ProjectRequest): Observable<Project> {
    return this.http.post<Project>(this.url, request);
  }

  update(id: string, request: ProjectRequest): Observable<Project> {
    return this.http.put<Project>(`${this.url}/${id}`, request);
  }

  archive(id: string): Observable<void> {
    return this.http.patch<void>(`${this.url}/${id}/archive`, {});
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`${this.url}/${id}`);
  }

  convertFromBid(bidId: string): Observable<Project> {
    return this.http.post<Project>(`${this.url}/convert-bid/${bidId}`, {});
  }
}

File: backend/src/main/java/com/resourceplanner/service/ProjectService.java
package com.resourceplanner.service;

import com.resourceplanner.dto.request.ProjectRequest;
import com.resourceplanner.dto.response.ProjectResponse;
import java.util.List;
import java.util.UUID;

public interface ProjectService {
    List<ProjectResponse> getAllActive();
    ProjectResponse getById(UUID id);
    ProjectResponse create(ProjectRequest request);
    ProjectResponse update(UUID id, ProjectRequest request);
    void archive(UUID id);
    void delete(UUID id);
    ProjectResponse convertFromBid(UUID bidId);
}
